src/components/timers/Tabata.tsx

```diff
@@ -1,3 +1,145 @@
-const Tabata = () => null;
+import '../../styles/tabata.css';
+import '../../styles/settings.css';
+import { useState, useEffect, useRef } from 'react';
+import { ButtonOptions } from '../generic/Buttons';
+import { TabataStats } from '../generic/TabataStats';
+
+/*
+  component description
+    - renders a Tabata timer that consists of a work period, a rest period, and set number of rounds
+    - user inputs for the work, rest, and rounds are embedded in the component rather than wrapped - opportunity for enhancement in the future as time permits
+    - several state variables to keep track of time elapsed, round, and shift in phase between work and rest
+    - some sharing across other functions / components but not as much as I would have liked - also another opportunity as time permits
+*/
+
+const Tabata = () => {
+    const [rounds, setRounds] = useState<number>(0);
+    const [workTarget, setWorkTarget] = useState<number>(0);
+    const [restTarget, setRestTarget] = useState<number>(0);
+    const [phase, setPhase] = useState<string>('');
+    const [isActive, setIsActive] = useState<boolean>(false);
+    const [isPaused, setIsPaused] = useState<boolean>(true);
+    const [roundsLeft, setRoundsLeft] = useState<number>(0);
+    const [secondsLeft, setSecondsLeft] = useState<number>(0);
+    const [cumulativeTime, setCumulativeTime] = useState<number>(0);
+
+    const secondsLeftRef = useRef(secondsLeft);
+    const phaseRef = useRef(phase);
+    const roundsLeftRef = useRef(roundsLeft);
+
+    useEffect(() => {
```

Comment:
It seems like this useEffect is solely keeping the ref and the state var in sync. If you get rid of the duplicative state/refs , you should be able to get rid of this useEffect as well